src/main/java/org/sagebionetworks/bridge/validators/SessionValidator.java

```diff
@@ -32,72 +88,169 @@ public void validate(Object obj, Errors errors) {
+                        long expMin = periodInMinutes(window.getExpiration());
                         if (expMin > intervalMin) {
-                            errors.rejectValue("expiration", "cannot be longer in duration than the session’s interval");
+                            errors.rejectValue(EXPIRATION_FIELD, EXPIRATION_LONGER_THAN_INTERVAL_ERROR);
                         }
                     }
                 }
                 errors.popNestedPath();
             }
+            validateTimeWindowOverlaps(errors, session);
         }
-        // Note however that an easy way to schedule a notification would be to create a session
-        // with notification info, but no assessments. So we might allow this at a later time.
         if (session.getAssessments().isEmpty()) {
-            errors.rejectValue("assessments", CANNOT_BE_NULL_OR_EMPTY);
+            errors.rejectValue(ASSESSMENTS_FIELD, CANNOT_BE_NULL_OR_EMPTY);
         } else {
             for (int i=0; i < session.getAssessments().size(); i++) {
                 AssessmentReference asmt = session.getAssessments().get(i);
                 
-                errors.pushNestedPath("assessments["+i+"]");
+                errors.pushNestedPath(ASSESSMENTS_FIELD+"["+i+"]");
                 if (isBlank(asmt.getGuid())) {
-                    errors.rejectValue("guid", CANNOT_BE_BLANK);
+                    errors.rejectValue(GUID_FIELD, CANNOT_BE_BLANK);
                 }
                 if (isBlank(asmt.getIdentifier())) {
-                    errors.rejectValue("identifier", CANNOT_BE_BLANK);
+                    errors.rejectValue(IDENTIFIER_FIELD, CANNOT_BE_BLANK);
                 }
                 if (isBlank(asmt.getAppId())) {
-                    errors.rejectValue("appId", CANNOT_BE_BLANK);
+                    errors.rejectValue(APP_ID_FIELD, CANNOT_BE_BLANK);
                 }
                 validateLabels(errors, asmt.getLabels());
-                validateColorScheme(errors, asmt.getColorScheme(), "colorScheme");
+                validateColorScheme(errors, asmt.getColorScheme(), COLOR_SCHEME_FIELD);
                 errors.popNestedPath();
             }
         }
+        for (int i=0; i < session.getNotifications().size(); i++) {
+            Notification notification = session.getNotifications().get(i);
+            
+            errors.pushNestedPath(NOTIFICATIONS_FIELD+"["+i+"]");
+            
+            if (notification.getNotifyAt() == null) {
+                errors.rejectValue(NOTIFY_AT_FIELD, CANNOT_BE_NULL);
+            }
+            validateFixedLengthPeriod(errors, notification.getOffset(), OFFSET_FIELD, true);
+            validateFixedLengthPeriod(errors, notification.getInterval(), INTERVAL_FIELD, false);
+            if (!session.getTimeWindows().isEmpty()) {
+                TimeWindow shortestWindow = shortestTimeWindow(session);
+                if (shortestWindow.getExpiration() != null) {
```

Comment:
We might have to check all of these, since it skips this validation if the window does not expire, which means the next longest window now becomes an issue and should be compared against the notification time.